tensorboard/lazy_test.py

```diff
@@ -87,6 +87,26 @@ def bad():
     with six.assertRaisesRegex(self, ValueError, expected_message):
       bad.day
 
+  def test_loads_only_once_even_when_result_equal_to_everything(self):
+    # This would fail if the implementation of `_memoize` used `==`
+    # rather than `is` to check for the sentinel value.
+    class EqualToEverything(object):
+      init_count = 0
```

Comment:
It seems a little better to just have a counter within `foo()` itself?  Since what we are really tracking is that `foo()` should be called just once, regardless of what it does.  That way we separate concerns so that EqualToEverything is only doing what it says on the label and not also being a counter.

This does require an indirection to get around python read-only closures, e.g. `foo.counter = getattr(foo, 'counter', 0) + 1`.

Reply:
I originally had the standard `count_box = [0]` trick, which I can
reinstate. You can’t just store the value on `foo` itself because that
will refer to the wrapped `LazyModule`; i.e., using

```python
    @lazy.lazy_load("foo")
    def foo():
      foo.counter = getattr(foo, "counter", 0) + 1
      return EqualToEverything()
```

raises “Circular import when resolving LazyModule 'foo'”. You can get around this with

```python
    @lazy.lazy_load("foo")
    def foo():
      foo.counter += 1
      return EqualToEverything()
    foo.counter = 0
```

but given that it’s this subtle I’m inclined to stick to the much
simpler close-over-`[0]`, which is at least somewhat idiomatic.

Reply:
Okay, I was being too clever.  The boxing singleton list is fine, sure.